cpp_modules/src/sc2_state/Sc2State.h

```diff
@@ -16,7 +17,7 @@ namespace Sc2 {
         int incomingVespeneCollectors = 0;
         int _populationLimit = 15;
         std::vector<Base> _bases = std::vector{Base()}; // (maybe) Replace with list
-        std::vector<Construction> _constructions{}; // Replace with list
+        std::list<Construction> _constructions{};
```

Comment:
Is there a reason why the constructions were changed to list, or was it just a design preference?

Reply:
There are many insertions and deletions made on _constructions. Insertions are always in the end, but the deletions are often in the middle of the vector.
Using a vector: Deletion of an element that is not in the end takes O(N) time. 
Using a list: Deletion of an element anywhere is constant.

Reply:
Makes a lot of sense. Thanks for the reply 😄